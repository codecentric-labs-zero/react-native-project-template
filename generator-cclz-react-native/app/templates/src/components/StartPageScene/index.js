import React from 'react'
import { View, Text } from 'react-native'

export default StartPageScene = (props) => (
  <View style={{flex: 1, flexDirection: 'column', justifyContent: 'center', alignItems: 'center'}}>
    <Text onPress={props.onButtonClick}>Navigate to next page</Text>
    <Text>Initial async state: {props.initialAsyncState}</Text>
  </View>
)
