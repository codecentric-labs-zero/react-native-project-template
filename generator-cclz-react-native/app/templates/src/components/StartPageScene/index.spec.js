import 'react-native';
import React from 'react';
import { Text } from 'react-native';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import StartPageScene from './index'

const textIs = (text) => (
  (e) => e.props().children === text
);

describe('StartPageScene', () => {

  it('renders correctly', () => {
    const tree = renderer.create(
      <StartPageScene />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls onButtonClick when clicking navigate to next page', () => {
    const onButtonClick = jest.fn();
    const element = shallow(<StartPageScene onButtonClick={onButtonClick} />);
    element.findWhere(textIs('Navigate to next page')).simulate('press');
    expect(onButtonClick).toBeCalled();
  });

  it('shows text for initialAsyncState', () => {
    const initialAsyncState = 'the state';
    const element = shallow(<StartPageScene initialAsyncState={initialAsyncState} />);
    expect(element.containsMatchingElement(<Text>Initial async state: {initialAsyncState}</Text>)).toBeTruthy();
  });

});
