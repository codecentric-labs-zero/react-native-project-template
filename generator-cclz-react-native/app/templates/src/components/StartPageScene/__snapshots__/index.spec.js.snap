exports[`StartPageScene renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={undefined}>
    Navigate to next page
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    Initial async state: 
  </Text>
</View>
`;
